modules/packages/LinearAlgebra.chpl

```diff
@@ -1450,37 +1450,55 @@ proc eig(A: [] ?t, param left = false, param right = false)
 
   if !left && !right {
     var vl, vr: [1..1, 1..n] t;
+    if t == complex{
+    LAPACK.geev(lapack_memory_order.row_major, 'N', 'N', copy, w, vl, vr);
+    var eigVals = w;
```

Comment:
Note that `eigVals` is bound to this scope and will not be valid at the return statement. You will instead need to declare `eigVals` and any other returns at the top of the function similar to `wr`and `wi`.

Reply:
@ben-albrecht but are we sure about the type of `eigVals` variable?

Depending on real or complex `t` it can be declared in two ways:
```
var eigVals: [1..n] t;
```
or 

```
var eigVals: [1..n] complex(numBits(t)*2);
```
as did in the `convertToCplx` proc.

Reply:
I think this should work:

```chpl
var eigVals: if t == complex then [1..n] t else [1..n] complex(numBits(t)*2);
```

Reply:
Thanks!